rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -923,6 +925,51 @@ protected int getReleaseRequestQueueSize() throws IOException {
         return getReleaseRequestQueue().getEntries().size();
     }
 
+    public boolean hasBridgeBtcTxSigHash(Sha256Hash sigHash) {
+        if (!activations.isActive(RSKIP379) || sigHash == null){
+            return false;
+        }
+
+        byte[] data = repository.getStorageBytes(
+            contractAddress,
+            getStorageKeyForBridgeBtcTxSigHash(sigHash)
+        );
+
+        return data != null &&
+           data.length == 1 &&
+           data[0] == TRUE_VALUE;
+    }
+
+    public void setBridgeBtcTxSigHash(Sha256Hash sigHash) {
+        if (!activations.isActive(RSKIP379) || sigHash == null) {
+            return;
+        }
+
+        if (hasBridgeBtcTxSigHash(sigHash)){
+            throw new IllegalStateException(String.format("Given bridge btc tx sigHash %s already exists in the index. Index entries are considered unique.", sigHash));
+        }
+
+        if (bridgeBtcTxSigHashes == null){
+            bridgeBtcTxSigHashes = new HashSet<>();
+        }
+
+        bridgeBtcTxSigHashes.add(sigHash);
+    }
+
+    private void saveBridgeBtcTxSigHashes() {
+        if (!activations.isActive(RSKIP379) || bridgeBtcTxSigHashes == null) {
+            return;
+        }
+
+        bridgeBtcTxSigHashes.forEach(bridgeBtcTxSigHash -> repository.addStorageBytes(
```

Comment:
Why Do We Try To Save The Same hashes Existing Already In **bridgeBtcTxSigHashes** Everytime We Call **saveBridgeBtcTxSigHashes**
Shouldn't That Throw Like An Error Or Something?